pkg/envoy/envoy.go

```diff
@@ -288,6 +297,10 @@ func (e *Envoy) StopEnvoy() error {
 	err := e.admin.quit()
 	if err != nil {
 		log.WithError(err).Fatal("Envoy: Admin quit failed, killing process ", e.cmd.Process.Pid)
+
+		e.cmdLocker.Lock()
```

Comment:
How can we make this Lock is ever Locked if it is held on line 161. The comment on line 185 states `We do not return after a successful start`. so the lock will be held until the process is running